tests/playwright/src/model/pages/create-machine-page.ts

```diff
@@ -0,0 +1,134 @@
+
+import { BasePage } from './base-page';
+import { ResourcesPage } from './resources-page';
+
+export class CreateMachinePage extends BasePage {
+  readonly heading: Locator;
+  readonly podmanMachineName: Locator;
+  readonly podmanMachineConfiguration: Locator;
+  readonly imagePathBox: Locator;
+  readonly browseImagesButton: Locator;
+  readonly podmanMachineCPUs: Locator;
+  readonly podmanMachineMemory: Locator;
+  readonly podmanMachineDiskSize: Locator;
+  readonly rootPriviledgesCheckbox: Locator;
+  readonly userModeNetworkingCheckbox: Locator;
+  readonly startNowCheckbox: Locator;
+  readonly closeButton: Locator;
+  readonly createMachineButton: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.heading = this.page.getByRole('heading', { name: 'Create Podman Machine' });
+    this.podmanMachineConfiguration = this.page.getByRole('form', { name: 'Properties Information' });
+    this.podmanMachineName = this.podmanMachineConfiguration.getByRole('textbox', { name: 'Name' });
+    this.imagePathBox = this.podmanMachineConfiguration.getByRole('textbox', { name: 'Image Path (Optional) ' });
+    this.browseImagesButton = this.podmanMachineConfiguration.getByRole('button', {
+      name: 'button-Image Path (Optional)',
+    });
+    this.podmanMachineCPUs = this.podmanMachineConfiguration.getByRole('slider', { name: 'CPU(s)' });
+    this.podmanMachineMemory = this.podmanMachineConfiguration.getByRole('slider', { name: 'Memory' });
+    this.podmanMachineDiskSize = this.podmanMachineConfiguration.getByRole('slider', { name: 'Disk size' });
+    this.rootPriviledgesCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', {
+      name: 'Machine with root privileges',
+    });
+    this.userModeNetworkingCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', {
+      name: 'User mode networking',
+    });
+    this.startNowCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', { name: 'Start the machine now' });
+    this.closeButton = this.page.getByRole('button', { name: 'Close' });
+    this.createMachineButton = this.page.getByRole('button', { name: 'Create' });
+  }
+
+  async createMachine(
+    machineName: string,
+    isRootful: boolean = true,
+    enableUserNet: boolean = false,
+    startNow: boolean = true,
+    setAsDefault: boolean = true,
+  ): Promise<ResourcesPage> {
+    await playExpect(this.podmanMachineConfiguration).toBeVisible();
+    await this.podmanMachineName.fill(machineName);
+
+    if (isRootful !== (await this.isEnabled(this.rootPriviledgesCheckbox))) {
+      await this.switchCheckbox(this.rootPriviledgesCheckbox);
+    }
+
+    if (enableUserNet !== (await this.isEnabled(this.userModeNetworkingCheckbox))) {
+      await this.switchCheckbox(this.userModeNetworkingCheckbox);
+    }
+
+    if (startNow !== (await this.isEnabled(this.startNowCheckbox))) {
+      await this.switchCheckbox(this.startNowCheckbox);
+    }
```

Comment:
These can be refactor in order for the if statements to not be required, there should be some method named `ensureCheckboxState()` that accepts the checkbox locator and the desired state of the checkbox which checks the initial state and switches the state in case it does not match the desired state.